### PR TITLE
Don't drop instance and job labels from the Prometheus receiver

### LINKED ISSUE #575: Prometheus Job and Instance labels
When using the Prometheus receiver and exporter I no longer get the `job` or `instance` labels that are default in Prometheus. Is this expected behaviour?

When Prometheus scrapes them I lose these fields even with `honor_labels: true`. I have manually checked the metrics and they are missing these labels so it's not an issue with Prometheus. Is there a way to manually add them back in with some `relabel_configs` magic? I have tried but it seems to get ignored whenever I use the `job` or `instance` labels.

I eventually want two exporters here one of which can be scraped locally by Prometheus and the other will forward onto another collecter. Our default use case is without Prometheus so scraping Prometheus with the collector isn't an option. 

Version: v0.2.3

```
    service:
      pipelines:
        metrics:
          receivers: [prometheus]
          exporters: [prometheus]
    exporters:
      prometheus:
        endpoint: "0.0.0.0:9090"  
    receivers:
      prometheus:
        config:
          global:
            scrape_interval: {{ .Values.scrapeInterval }}
          scrape_configs:
          - job_name: 'pilot'
            kubernetes_sd_configs:
            - role: endpoints
              namespaces:
                names:
                - {{ .Release.Namespace }}
            relabel_configs:
            - source_labels: [__meta_kubernetes_service_name, __meta_kubernetes_endpoint_port_name]
              action: keep
              regex: istio-pilot;http-monitoring
```

```
- job_name: 'oc-agent'
      scrape_interval: 5s
      honor_labels: true
      kubernetes_sd_configs:
      - role: endpoints
        namespaces:
          names:
          - {{ .Release.Namespace }}
      relabel_configs:
      - source_labels: [__meta_kubernetes_service_name, __meta_kubernetes_endpoint_port_name]
        action: keep
        regex: otel-collector;prometheus
```

### LINKED ISSUE #2363: Automatically generated metrics and labels by prometheus are dropped by the prometheus receiver
**Describe the bug**
Labels and metrics that are automatically generated by prometheus are dropped by the prometheus receiver. Example - metric `up and scrape_duration_seconds (few other scape_ metrics)` and labels `job and instance`. Is this intentional or a potential bug?

**Steps to reproduce**
I set up a pipeline with prometheus receiver and prometheusremotewrite. Config file is provided below.

**What did you expect to see?**
I expected to see the metrics and labels mentioned [here](https://prometheus.io/docs/concepts/jobs_instances/#automatically-generated-labels-and-time-series)

**What did you see instead?**
Above metrics and labels were dropped.

**What version did you use?**
Version: 0.17.0

**What config did you use?**
```
receivers:
  prometheus:
    config:
      scrape_configs:
        - job_name: 'otel-collector'
          scrape_interval: 10s
          static_configs:
            - targets: [ '0.0.0.0:8888' ]

exporters:
  prometheusremotewrite:
    endpoint: "http://localhost:1234/receive"

service:
  pipelines:
    metrics:
      receivers: [prometheus]
      exporters: [prometheusremotewrite]

```

**Additional context**
I briefly looked into the code and I observed below

- Prometheus generated labels seems to be dropped [here](https://github.com/open-telemetry/opentelemetry-collector/blob/release/v0.17.x/receiver/prometheusreceiver/internal/metricsbuilder.go#L165-L168)
- Prometheus generated metrics seems to be dropped [here](https://github.com/open-telemetry/opentelemetry-collector/blob/release/v0.17.x/receiver/prometheusreceiver/internal/metricsbuilder.go#L96-L114)

### LINKED ISSUE #2499: Prometheus receiver: Move job and instance labels into resource attributes
**Is your feature request related to a problem? Please describe.**
Prometheus adds the labels _job_ and _instance_ to identify the target scraped - this maps to the Resource in opentelemetry against which the metrics are associated with. Looks like the Resource is currently empty, we could populate these two labels as attributes under Resource.

**Describe the solution you'd like**
Move (or copy) the two labels _job_ and _instance_ as attributes under Resource.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by rakyll**
Fixes #575
Fixes #2499
Fixes  #2363
Fixes  open-telemetry/wg-prometheus#37
Fixes  open-telemetry/wg-prometheus#39
Fixes open-telemetry/wg-prometheus#44

Passing compliance tests:

$ go test --tags=compliance  -run  "TestRemoteWrite/otelcollector/Job.+" -v ./
=== RUN   TestRemoteWrite
=== RUN   TestRemoteWrite/otelcollector
=== RUN   TestRemoteWrite/otelcollector/JobLabel
=== PAUSE TestRemoteWrite/otelcollector/JobLabel
=== CONT  TestRemoteWrite/otelcollector/JobLabel
--- PASS: TestRemoteWrite (10.02s)
    --- PASS: TestRemoteWrite/otelcollector (0.00s)
        --- PASS: TestRemoteWrite/otelcollector/JobLabel (10.02s)
PASS
ok  	github.com/prometheus/compliance/remote_write	10.382s
$  go test --tags=compliance  -run  "TestRemoteWrite/otelcollector/Instance.+" -v ./
=== RUN   TestRemoteWrite
=== RUN   TestRemoteWrite/otelcollector
=== RUN   TestRemoteWrite/otelcollector/InstanceLabel
=== PAUSE TestRemoteWrite/otelcollector/InstanceLabel
=== CONT  TestRemoteWrite/otelcollector/InstanceLabel
--- PASS: TestRemoteWrite (10.01s)
    --- PASS: TestRemoteWrite/otelcollector (0.00s)
        --- PASS: TestRemoteWrite/otelcollector/InstanceLabel (10.01s)
PASS
ok  	github.com/prometheus/compliance/remote_write	10.291s
$ go test --tags=compliance  -run  "TestRemoteWrite/otelcollector/RepeatedLabels.+" -v ./
=== RUN   TestRemoteWrite
=== RUN   TestRemoteWrite/otelcollector
--- PASS: TestRemoteWrite (0.00s)
    --- PASS: TestRemoteWrite/otelcollector (0.00s)
testing: warning: no tests to run
PASS

### CODE REVIEW

**Comment by rakyll** [comment]
cc @odeke-em

**Review by odeke-em — APPROVED** [review]
This is quite the awesome PR!! Thank you @rakyll , LGTM!

**Comment by rakyll** [comment]
Will fix the tests soon.

**Comment by rakyll** [comment]
The failing test (TestEndToEndSummarySupport) has a hard dependency on the receiver and byte to byte comparing the output. Is there a way we can remove this test and add something a bit more self contained @odeke-em?

**Comment by odeke-em** [comment]
Please t.Skipf("@odeke-em please take a look") it and then we'll be gucci to go.

**Comment by rakyll** [comment]
It seems like the pull based Prometheus exporter is not exposing the job and instance labels and it seems like it's a problem itself.

**Comment by rakyll** [comment]
I can't see why build-and-test  is failed and can't reproduce it myself locally. Can someone paste the log?

**Comment by alolita** [comment]
Nicely done @rakyll ty!

**Comment by rakyll** [comment]
cc @bogdandrutu @tigrannajaryan Can you review this? It's required for the stability.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
I don't know enough about this receiver to understand the change fully. @bogdandrutu @jrcamp maybe you can help review?

File: receiver/prometheusreceiver/metrics_receiver_test.go [hunk 1]
```diff
@@ -263,12 +270,17 @@ func verifyTarget1(t *testing.T, td *testData, mds []internaldata.MetricsData) {
 			Name:        "go_threads",
 			Description: "Number of OS threads created",
 			Type:        metricspb.MetricDescriptor_GAUGE_DOUBLE,
+			LabelKeys:   []*metricspb.LabelKey{{Key: "instance"}, {Key: "job"}},
```
Comment (dashpole) [inline receiver/prometheusreceiver/metrics_receiver_test.go:273]: Should `instance` and `job` be part of the resource instead of labels?

**Comment by Aneurysm9** [comment]
How does this relate to #2683?

File: receiver/prometheusreceiver/metrics_receiver_test.go (hunk 1 above)
Comment (rakyll) [reply receiver/prometheusreceiver/metrics_receiver_test.go:273]: You are correct, working on it and will send another version.

**Comment by rakyll** [comment]
Moving this PR to https://github.com/open-telemetry/opentelemetry-collector/pull/2979. This issue requires an exporter change which populates the "instance" and "job" from resource attributes.

**Comment by rakyll** [comment]
@Aneurysm9, just seeing your last comment, #2683 is related to this but it seems like #2897 was doing the right thing to turn these labels into resource attributes. What was missing was to take them from the resource attributes and produce Prometheus labels in the exporters which #2979 is doing.

**Pull request closed without merging**